src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -3159,3 +3178,75 @@ void Pane::CollectTaskbarStates(std::vector<winrt::TerminalApp::TaskbarState>& s
         _secondChild->CollectTaskbarStates(states);
     }
 }
+
+void Pane::EnableBroadcast(bool enabled)
+{
+    if (_IsLeaf())
+    {
+        _broadcastEnabled = enabled;
+        UpdateVisuals();
+    }
+    else
+    {
+        _firstChild->EnableBroadcast(enabled);
+        _secondChild->EnableBroadcast(enabled);
+    }
+}
+
+void Pane::BroadcastKey(const winrt::Microsoft::Terminal::Control::TermControl& sourceControl,
+                        const WORD vkey,
+                        const WORD scanCode,
+                        const winrt::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                        const bool keyDown)
+{
+    WalkTree([&](auto pane) {
+        if (pane->_IsLeaf() && pane->_control != sourceControl && !pane->_control.ReadOnly())
+        {
+            pane->_control.RawWriteKeyEvent(vkey, scanCode, modifiers, keyDown);
+        }
+    });
+}
+
+void Pane::BroadcastChar(const winrt::Microsoft::Terminal::Control::TermControl& sourceControl,
+                         const wchar_t character,
+                         const WORD scanCode,
+                         const winrt::Microsoft::Terminal::Core::ControlKeyStates modifiers)
+{
+    WalkTree([&](auto pane) {
+        if (pane->_IsLeaf() && pane->_control != sourceControl && !pane->_control.ReadOnly())
+        {
+            pane->_control.RawWriteChar(character, scanCode, modifiers);
+        }
+    });
+}
+
+void Pane::BroadcastString(const winrt::Microsoft::Terminal::Control::TermControl& sourceControl,
+                           const winrt::hstring& text)
+{
+    WalkTree([&](auto pane) {
+        if (pane->_IsLeaf() && pane->_control != sourceControl && !pane->_control.ReadOnly())
+        {
+            pane->_control.RawWriteString(text);
+        }
+    });
+}
+
+void Pane::_ControlReadOnlyChangedHandler(const winrt::Windows::Foundation::IInspectable& /*sender*/, const winrt::Windows::Foundation::IInspectable& /*e*/)
+{
+    UpdateVisuals();
+}
+
+winrt::Windows::UI::Xaml::Media::SolidColorBrush Pane::_ComputeBorderColor()
+{
+    if (_lastActive)
+    {
+        return s_focusedBorderBrush;
+    }
+
+    if (_broadcastEnabled)
+    {
+        return s_broadcastBorderBrush;
+    }

```

Comment:
Should we not set the color to `s_broadcastBorderBrush` if the pane is in read only mode? It might be confusing since the broadcasted input won't be sent to it but it has the same border colors as the others.